backend/src/main/java/org/atlantbh/internship/auctionapp/service/impl/BidServiceImpl.java

```diff
@@ -26,7 +37,26 @@ public ProductBidResponse getProductBidInfo(Long productId) {
     }
 
     @Override
-    public List<UserProfileProductsInfo> getBidsForUser(Long userId) {
+    public List<ProductBidsInfo> getBidsForUser(Long userId) {
         return bidRepository.getUserBids(userId);
     }
+
+    @Override
+    public SuccessfulAndMessage bid(BidRequest bidRequest, PersonDetails user) throws BadRequestException {
+        ProductEntity product = productRepository.findById(bidRequest.getProductId())
+                .orElseThrow(() -> new BadRequestException("Product with given id does not exist."));
+        if (product.getUser().getId().equals(user.getId()))
+            throw new BadRequestException("You can't bid to your own products");
+        BigDecimal currentHighest = bidRepository.findFirst1ByProductIdOrderByPriceDesc(bidRequest.getProductId())
+                .map(BidEntity::getPrice).orElse(product.getStartingPrice());
```

Comment:
Declare variables right before they're used. There are two if statements below that do not use this variable and if they're fulfilled we just wasted time computing this value.